subsys/random/rand32_timer.c

```diff
@@ -67,7 +67,14 @@ void z_impl_sys_rand_get(void *dst, size_t outlen)
 	uint32_t ret;
 
 	while (outlen) {
-		ret = sys_rand32_get();
+		/*
+		 * Parameters: (rand_val * multiplier + increment) & modulus
+		 *
+		 * Increment value must be odd and coprime with multiplier.
+		 * Modulus performs modulo 2^32 operation.
+		 */
+		ret = ((sys_rand32_get() * 32310901L + 123456789L) & 0xffffffff);
```

Comment:
It is unclear to me how this would affect the problem. As I understand, the test is catching duplicates being returned from the random function (due to the weak time-based generator), performing a linear operation on the output of the random function is going to result in exactly the same collisions, as the colliding results from `sys_rand32_get()` will still collide after arithmetic is performed on them.

Reply:
@d3zd3z, if there is a desire to run these tests on targets with bad (no) entropy source, do you think that adding a static incrementing variable with each request would at least patch over the problem? 

static uint32_t zork;
....
ret = sys_rand32_get() + (zork++);

We don't want to encourage usage of rand32_timer and only left it in for testing purposes. If it still causes problems then make a minor adjustment to avoid back-to-back duplicates or block the target from this test case.

Reply:
@dleach02 if I'm not mistaken, this is already implemented in the timer-based PRNG here: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/random/rand32_timer.c#L31-L47, as the value of  `_rand32_counter` is incremented by `_RAND32_INC` every time `sys_rand32_get` function is called.

Reply:
Ah, I was not aware of that change. Thank you for pointing it out.